### PR TITLE
[PROTOCOL-350] and [348]. Moving accrued interest on-chain

### PR DESCRIPTION
**Pull request by hensha256**

### CODE REVIEW

File: contracts/base/LendingPool.sol [hunk 1]
```diff
@@ -64,62 +66,78 @@ contract LendingPool is Base, LendingPoolInterface {
...
         isInitialized()
         whenNotPaused()
         whenLendingPoolNotPaused(address(this))
     {
+        // Update the exchange rate as the tokens in compound will have gained interest
+        _updateExchangeRate();
+
+        uint256 tTokenAmount = lendingTokenAmount.mul(EXCHANGE_RATE_DECIMALS).div(
+            exchangeRate
+        );
```
Comment (passabilities) [inline contracts/base/LendingPool.sol:82]: why are we multiplying by 18? did you mean `10**18`?

File: contracts/base/LendingPool.sol [hunk 2]
```diff
@@ -266,33 +255,73 @@ contract LendingPool is Base, LendingPoolInterface {
...
     }
 
     /** Internal functions */
 
+    function _updateExchangeRate() internal {
+        // calculate the total lendingToken in the protocol (on loan + not on loan)
+        MarketStateLib.MarketState memory lendingTokenMarket = _markets().getGlobalMarket(
+            address(4)
+        );
```
Comment (passabilities) [inline contracts/base/LendingPool.sol:290]: whats this?

File: contracts/base/LendingPool.sol [hunk 3]
```diff
@@ -266,33 +255,73 @@ contract LendingPool is Base, LendingPoolInterface {
...
+            );
+        }
+
+        uint256 totalLendingToken = totalOnLoan.add(totalNotOnLoan);
+
+        // fetch total TToken for this lending token
+        uint256 totalTToken = tToken.totalSupply();
+
+        // exchange rate = lending tokens per ttoken (with EXCHANGE_RATE_DECIMALS)
+        exchangeRate = totalLendingToken.mul(EXCHANGE_RATE_DECIMALS).div(totalTToken);
```
Comment (passabilities) [inline contracts/base/LendingPool.sol:322]: again here

File: contracts/interfaces/TTokenInterface.sol [hunk 4]
```diff
@@ -10,6 +10,12 @@ import "@openzeppelin/contracts-ethereum-package/contracts/token/ERC20/IERC20.so
  * @author develop@teller.finance
  */
 contract TTokenInterface is IERC20 {
+    /**
+     * @notice function to fetch the number of decimal places a TToken has
+     * @return the number of decimals
+     */
+    function decimals() external view returns (uint8);
+
```
Comment (passabilities) [inline contracts/interfaces/TTokenInterface.sol:18]: could just make this extend the new `IERC20Detailed` Interface instead

File: test/utils/logicNames.js [hunk 5]
```diff
@@ -4,7 +4,6 @@ module.exports = {
     TokenCollateralLoans: 'TokenCollateralLoans',
```
Comment (passabilities) [inline test/utils/logicNames.js:4]: above this line we can also remove the `Lenders`

**Review by passabilities — CHANGES_REQUESTED** [review]
Looks good and definitely cleans up the code base as well! 

NOTE: Please rebase this or merge in the branch `feature/merge-LP-MS` as I get rid of the `MarkesState` contract and merge the functionality directly into the `LendingPool`

**Review by tpscrpt — APPROVED** [review]

**Review by salazarguille — CHANGES_REQUESTED** [review]
Great job! @hensha256 
Just some comments/questions
- 2 tests are failing.
- are the integration tests passing?

**Review by passabilities — DISMISSED** [review]

**Review by passabilities — APPROVED** [review]

**Pull request merged**